src/rpc/types/mod.rs

```diff
@@ -164,6 +166,35 @@ pub struct ApiTipsetKey(pub Option<TipsetKey>);
 /// See: <https://github.com/filecoin-project/lotus/issues/11461>.
 #[derive(Debug, Eq, PartialEq, Clone)]
 pub struct AddressOrEmpty(pub Option<Address>);
+#[derive(Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "PascalCase")]
+pub struct ClaimLotusJson {
+    // The provider storing the data (from allocation).
+    #[schemars(with = "u64")]
+    pub provider: ActorID,
+    // The client which allocated the DataCap (from allocation).
+    #[schemars(with = "u64")]
+    pub client: ActorID,
+    // Identifier of the data committed (from allocation).
+    #[schemars(with = "LotusJson<Cid>")]
+    #[serde(with = "crate::lotus_json")]
+    pub data: Cid,
+    // The (padded) size of data (from allocation).
+    #[schemars(with = "u64")]
+    pub size: PaddedPieceSize,
+    // The min period after term_start which the provider must commit to storing data
+    #[schemars(with = "u64")]
+    pub term_min: ChainEpoch,
+    // The max period after term_start for which provider can earn QA-power for the data
+    #[schemars(with = "u64")]
+    pub term_max: ChainEpoch,
+    // The epoch at which the (first range of the) piece was committed.
+    #[schemars(with = "u64")]
+    pub term_start: ChainEpoch,
+    // ID of the provider's sector in which the data is committed.
+    #[schemars(with = "u64")]
```

Comment:
Why is it required to put the type `#[schemars(with = "u64")]` everywhere? What's the behaviour, and reason, when the types don't match? For example, in `ChainEpoch` you put `u64` whereas it's `i64` underneath. What will happen for values like `-1`?

Reply:
It needs to be fixed I’m pretty sure.

Reply:
I've taken a look at this - it's not needed at all, except for enums with `repr(u64)`. I'll remove all the ocurrences.

Reply:
done

Reply:
Aha, it was needed for `u64` wrappers, like `PaddedPieceSize`.

Reply:
Basically the tactic here is as follows - try compiling without any attributes, see what breaks, fix that.
This whole schemars thing is a bit confusing until one gets used to it.